client/src/components/ExpandedRowDataEdit.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {
+    MouseEvent,
+    useRef,
+    useState
+} from 'react';
+import type {DataEditProps} from './ExpandedRow';
+import {fetchData} from '../services/fetch';
+import {logError} from '../services/log';
+
+const ExpandedRowDataEdit = ({data, onError, onUpdate}: DataEditProps) => {
+    // This reference object is use to collect updated data by individual form input control.
+    const collectedData = useRef({...data});
+
+    // Use this state to track and temporarily disabled all buttons when Save button is clicked.
+    const [isSaving, setIsSaving] = useState(false);
+
+    const handleSaveClicked = ({currentTarget}: MouseEvent<HTMLButtonElement>) => {
+        if (JSON.stringify(collectedData.current) !== JSON.stringify(data)) {
+            const {note, jiraTicket} = collectedData.current;
+
+            setIsSaving(true);
+
+            fetchData(`/api/action/save/${data.id}`, {
```

Comment:
In a future iteration, this feels like a place to use Redux or React Query or some kind of internal state management that controls how often you make calls to the API so that you don't have race conditions if the button is clicked more than once or the component unmounts.